torch/_tensor_docs.py

```diff
@@ -6362,6 +6375,23 @@ def callable(a, b) -> number
 masked_scatter(mask, tensor) -> Tensor
 
 Out-of-place version of :meth:`torch.Tensor.masked_scatter_`
+
+.. note::
+
+    The :attr:`self` tensor will be broadcast to the shape of the
+    :attr:`mask` tensor, according to 
+    :ref:`broadcasting semantics <broadcasting-semantics>`.
```

Comment:
So self broadcasts over mask. Is there also broadcasting between mask and source? I would expect that the result size would be the result of broadcasting the three tensors

Reply:
Based on my understanding, I don't believe there is a broadcast between `mask` and `source`. It seems that the dimensions of `source` are not considered at all in the operation, except for the requirement that `source` must have at least as many elements as `mask` has `1`s.

For example, here `mask` and `source` have non-broadcastable dimensions, but `masked_scatter` works fine:
```python
mask = torch.tensor([[0, 0, 0, 1, 1],
                     [1, 1, 0, 1, 1],
                     [1, 1, 1, 0, 1],
                     [1, 1, 1, 1, 1]])
source = torch.tensor([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14])
output = self.masked_scatter(mask, source)
```

Output:
```
tensor([[ 0,  0,  0,  0,  1],
        [ 2,  3,  0,  4,  5],
        [ 6,  7,  8,  0,  9],
        [10, 11, 12, 13, 14]])
```

Whereas the simple broadcast operation fails:
```python
mask = torch.tensor([[0, 0, 0, 1, 1],
                     [1, 1, 0, 1, 1],
                     [1, 1, 1, 0, 1],
                     [1, 1, 1, 1, 1]])
source = torch.tensor([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14])
output = source + mask
```

Output:
```
RuntimeError: The size of tensor a (5) must match the size of tensor b (15) at non-singleton dimension 2
```

Although I'm fully willing to admit that I may be missing something here 🙂

Reply:
Right, we don't have broadcasting over `source`, but we do over the other two arguments.
```python
>>> import torch
>>> self = torch.randn(1, 5)
>>> mask = torch.tensor([0, 1, 1, 0, 0]).unsqueeze(1)
>>> source = torch.arange(10, dtype=torch.float)
>>> self.masked_scatter(mask, source)
tensor([[ 0.1844,  2.8943, -1.1053,  1.7455,  0.0311],
        [ 0.0000,  1.0000,  2.0000,  3.0000,  4.0000],
        [ 5.0000,  6.0000,  7.0000,  8.0000,  9.0000],
        [ 0.1844,  2.8943, -1.1053,  1.7455,  0.0311],
        [ 0.1844,  2.8943, -1.1053,  1.7455,  0.0311]])
```

Broadcasting is an operation that takes two shapes and computes "the supremum" of them. If it works for one input wrt. another, it will work for the second one wrt. the first one (or it should at least!)

Reply:
Ahh gotcha - thanks for the clarification! I'll go ahead and re-word this to make it clear that the broadcast isn't a one-way street